https://issues.redhat.com/browse/ACM-9551

dockerymick opened the pull request:
https://issues.redhat.com/browse/ACM-9551

dockerymick committed:
https://issues.redhat.com/browse/ACM-9551

swopebe commented on governance/generate_pol_operator_install.adoc:
```diff
@@ -0,0 +1,152 @@
+[#policy-gen-install-operator]
```
Can we make this file name and id clear that it is for compliance operator?

something like [generate_compliance_install.adoc]

swopebe commented on governance/generate_pol_operator_install.adoc:
```diff
@@ -0,0 +1,152 @@
+Complete the following steps:
+
+. Create a YAML file with a `Namespace`, a `Subscription`, and an `OperatorGroup` manifest called `compliance-operator.yaml`. The following example installs these manifests in the `compliance-operator` namespace:
++
```
this YAML seems off in the PR, may need an extra space or something.

swopebe replied:
yeah, just needed a space, it seems right now.

swopebe committed:
Update generate_pol_operator_install.adoc

check YAML

swopebe commented on governance/generate_pol_operator_install.adoc:
```diff
@@ -0,0 +1,153 @@
+----
+
+. Create a `PolicyGenerator` configuration file. View the following `PolicyGenerator` policy example that installs the Compliance Operator on all {ocp-short} managed clusters:
++
```
same here, so just create a space before the YAML.

dockerymick replied:
I checked out my branch to view through VSC, it seems like the new line is not required. Here are the screen captures. Please let me know if i am missing something
<img width="1438" alt="No new line" src="https://github.com/stolostron/rhacm-docs/assets/60616885/2920d1c0-d4a5-4ebe-90e0-fb089d2f6167">
<img width="1429" alt="With new line" src="https://github.com/stolostron/rhacm-docs/assets/60616885/eebd6819-7fb0-4af6-bdff-c719c5ba4b51">

swopebe commented on governance/generate_pol_operator_install.adoc:
```diff
@@ -0,0 +1,153 @@
+----
+
+. Add the policy generator to your `kustomization.yaml` file. The `generators` section might resemble the following configuration:
++
```
and here

gparvin commented on gitops/main.adoc:
```diff
@@ -7,6 +7,6 @@ include::gitops_console.adoc[leveloffset=+2]
 include::gitops_push_pull.adoc[leveloffset=+2]
-include::gitops_policy_operator.adoc[leveloffset=+2]
 include::gitops_manage_policy_def.adoc[leveloffset=+2]
+include::gitops_policy_operator.adoc[leveloffset=+2]
```
In ACM 2.10 we are adding a Tech Preview feature called the "Operator Policy".  This filename is going to be confused with that new feature.  Since this documentation is focused on the policy generator feature, can you make sure "generator" is included in the filenames?

dockerymick replied:
```suggestion
include::gitops_policy_generator.adoc[leveloffset=+2]
```

dockerymick commented on governance/generate_pol_operator_install.adoc:
```diff
@@ -0,0 +1,153 @@
+[#policy-gen-install-operator]
```
```suggestion
[#install-compliance-operator-policy]
```

dockerymick commented on governance/generate_pol_operator_install.adoc:
```diff
@@ -0,0 +1,153 @@
+          severity: low
+----
+
+//clarify that these should be committed to git. Are there verification steps? | MJ | 01/31/24
```
@gparvin

dockerymick replied:
Thanks for your approval Gus! Will you help with adding a verification step if there is one?

gparvin replied:
I think the content after: "As a result, the generated policy resembles the following file:" is what I would consider the verification step in the doc.  I did run through the yaml and verify the content previously.  I can re-check it too if you're concerned the yaml might not be right.

dockerymick replied:
@gparvin excuse the delay. Yes, will you share the file that will be displayed to the user?

dockerymick committed:
updating file

dockerymick committed:
Update governance/generate_pol_operator_install.adoc

dockerymick committed:
Update gitops/main.adoc

dockerymick committed:
Updating the file name

dockerymick committed:
Making more updates

gparvin commented on governance/generate_pol_operator_install.adoc:
```diff
@@ -0,0 +1,152 @@
+  name: install-compliance-operator
+policyDefaults:
+  namespace: policies
+  placement:
+    clusterSelectors:
+      vendor: "OpenShift"
```
Because we've been switching from PlacementRule to Placement, could you replace these lines with:
```
  placement:
    labelSelector:
      matchExpressions:
        - key: vendor
          operator: In
          values:
            - "OpenShift"
```

gparvin commented on governance/generate_pol_operator_install.adoc:
```diff
@@ -0,0 +1,152 @@
++
+[source,yaml]
+----
+apiVersion: apps.open-cluster-management.io/v1
+kind: PlacementRule
+metadata:
+  name: placement-install-compliance-operator
+  namespace: policies
+spec:
+  clusterConditions:
+    - status: "True"
+      type: ManagedClusterConditionAvailable
+  clusterSelector:
+    matchExpressions:
+      - key: vendor
+        operator: In
+        values:
+          - OpenShift
```
After switching to Placement in the previous comment, this PlacementRule resource becomes the following Placement resource:
```
apiVersion: cluster.open-cluster-management.io/v1beta1
kind: Placement
metadata:
  name: placement-install-compliance-operator
  namespace: policies
spec:
  predicates:
  - requiredClusterSelector:
      labelSelector:
        matchExpressions:
        - key: vendor
          operator: In
          values:
          - OpenShift
```

gparvin commented on governance/generate_pol_operator_install.adoc:
```diff
@@ -0,0 +1,152 @@
+  name: binding-install-compliance-operator
+  namespace: policies
+placementRef:
+  apiGroup: apps.open-cluster-management.io
+  kind: PlacementRule
+  name: placement-install-compliance-operator
```
The binding is also slightly different due to the placement switch:
```
  apiGroup: cluster.open-cluster-management.io
  kind: Placement
  name: placement-install-compliance-operator
```

gparvin commented:
We are trying to make sure GRC content is consistently using the cluster placement so I added some comments around that.  Thanks

dockerymick committed:
Update generate_pol_operator_install.adoc

Updates based on dev feedback

dockerymick commented:
@gparvin thanks for your patience and feedback. I have addressed your comments. When you have an opportunity, will you review my comment here: https://github.com/stolostron/rhacm-docs/pull/6087/files#r1481511400

gparvin reviewed (approved):
Thanks, in the next release we will want to switch this to use the new Operator Policy features we are creating.  This looks good now.

dockerymick commented on governance/generate_pol_operator_install.adoc:
```diff
@@ -0,0 +1,154 @@
+          severity: low
+----
+
+//clarify that these should be committed to git. Are there verification steps? | MJ | 01/31/24
```
```suggestion
As a result, the generated policy is displayed.
```

dockerymick committed:
Update governance/generate_pol_operator_install.adoc
